caiman/source_extraction/volpy/volpy_gui.py

```diff
@@ -0,0 +1,273 @@
+                 {'name': 'REMOVE', 'type': 'action'}, 
+                 {'name': 'SHOW ALL', 'type': 'action'},
+                 {'name': 'CLEAR', 'type': 'action'}, 
+                 {'name': 'IMAGES', 'type': 'list', 'values': ['MEAN','CORR']},
+                 {'name': 'MODE', 'type': 'list', 'values': ['POLYGON','CELL MAGIC WAND']},
+                 {'name': 'MAGIC WAND PARAMS', 'type': 'group', 'children': [{'name': 'MIN RADIUS', 'type': 'int', 'value': 4},
+                                                                    {'name': 'MAX RADIUS', 'type': 'int', 'value': 10},
+                                                                    {'name': 'ROUGHNESS', 'type': 'int', 'value': 1}]}]
+
+pars_action = Parameter.create(name='params_action', type='group', children=params_action) 
+
+neuron_action.setParameters(pars_action, showTop=False)
+neuron_action.setWindowTitle('Parameter Action')
+mode = pars_action.getValues()['MODE'][0]
+    
+def mouseClickEvent(event):
+    global mode, x, y, i, j, pts, roi, cur_image
+    if mode == "POLYGON":
+        pos = img.mapFromScene(event.pos())
+        x = int(pos.x())
+        y = int(pos.y())
+        j, i = pos.y(), pos.x()
+        i = int(np.clip(i, 0, dims[0] - 1))
+        j = int(np.clip(j, 0, dims[1] - 1))
+        p1.plot(x=[i], y=[j], symbol='o', pen=None, symbolBrush='y', symbolSize=5)#, symbolBrush=pg.intColor(i,6,maxValue=128))
+        pts.append([i, j])
+    elif mode == "CELL MAGIC WAND":
+        p1.clear()
+        p1.addItem(img)
+        pts = []
+        pos = img.mapFromScene(event.pos())
+        try:
+            x = int(pos.x())
+            y = int(pos.y())
+            j, i = pos.y(), pos.x()
+            p1.plot(x=[i], y=[j], symbol='o', pen=None, symbolBrush='y', symbolSize=5)#, symbolBrush=pg.intColor(i,6,maxValue=128))
+            min_radius = pars_action.getValues()['MAGIC WAND PARAMS'][1]['MIN RADIUS'][0]
+            max_radius = pars_action.getValues()['MAGIC WAND PARAMS'][1]['MAX RADIUS'][0]
+            roughness = pars_action.getValues()['MAGIC WAND PARAMS'][1]['ROUGHNESS'][0]
+            roi, edge = cell_magic_wand_single_point(adjust_contrast(cur_img.copy(), hist.getLevels()[0], hist.getLevels()[1]), (j, i), 
+                                               min_radius=min_radius, max_radius=max_radius, 
+                                               roughness=roughness, zoom_factor=1)
+            ROI8 = np.uint8(roi * 255)
+            contours = cv2.findContours(cv2.threshold(ROI8, 100, 255, 0)[1], cv2.RETR_TREE,
+                                             cv2.CHAIN_APPROX_NONE)[0][0][:,0,:]
+            pts = contours.tolist()
+            pp = pts.copy()
+            pp.append(pp[0])
+            p1.plot(x=np.array(pp)[:,0], y=np.array(pp)[:,1], pen=pen)
+            roi = roi * 1.
+            #plt.imshow(roi)
+            #plt.scatter(contours[:,0], contours[:,1])
+        except:
+            pass
```

Comment:
If the try block fails, might it be more sensible to terminate the program?

Reply:
Sometimes if you click on certain points, cell magic wand will give you bug. We just ignore that and select another point.